sros2/sros2/api/_utilities.py

```diff
@@ -87,7 +87,7 @@ def build_key_and_cert(subject_name, *, ca=False, ca_key=None, issuer_name=''):
         ).serial_number(
             x509.random_serial_number()
         ).not_valid_before(
-            utcnow
+            utcnow - datetime.timedelta(days=1)
```

Comment:
This feels hacky. Can we localize this? Even starting at midnight that day feels a little nicer, although is arguably just as arbitrary :stuck_out_tongue: .

Reply:
That's literally how it was before the refactor https://github.com/ros2/sros2/pull/138 :wink: 

https://github.com/ros2/sros2/blame/9f22e40b25b7c1486fbbc6cfdfdb19267ba3e05b/sros2/sros2/api/__init__.py#L236

Reply:
I'm against this hack on principle.
Folks should really fix/synchronize there time sources.
I'm not sure what localizing it would do, given timezones are always changing in terms of geography and UTC offset dates.

Reply:
> That's literally how it was before the refactor #138 :wink:

Ha! Some genius thought it would be a good idea to throw away that hack.

Reply:
@mikaelarguedas what is the actual breakage observed in connext?

Reply:
> Ha! Some genius thought it would be a good idea to throw away that hack innocent .

At least your consistent in your perception of the hack :wink: 

> what is the actual breakage observed in connext?

Oh my bad, it came up here https://github.com/ros2/ci/pull/436#issuecomment-624874296

Reply:
> I'm against this hack on principle.
Folks should really fix/synchronize there time sources.

I'm in agreement with that.

The one thing that's not clear to me is if it's a machine time sync issue or a connext issue that's not parsing it correctly.
The spec seems pretty clear about it though:
```xml
<!-- Format is CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm] The time zone may
be specified as Z (UTC) or (+|-)hh:mm. Time zones that aren't
specified are considered UTC.
-->
<not_before>2013-10-26T00:00:00</not_before>
<not_after>2018-10-26T22:45:30</not_after>
```

Reply:
> The one thing that's not clear to me is if it's a machine time sync issue or a connext issue that's not parsing it correctly.

Time seems ok. If it's off, it couldn't be more than a few minutes.

Time in `permissions.p7s` (time when I built test_security?)

```xml
      <validity>^M
        <not_before>2020-05-05T21:57:05</not_before>^M
        <not_after>2030-05-04T21:57:05</not_after>^M
      </validity>^M
```

Time on mini3 after running `test_security` tests

```console
$ date
Wed May  6 15:02:58 PDT 2020
```

Current time UTC after formatting the above text 22:05 https://www.timeanddate.com/worldclock/timezone/utc

Reply:
thanks for the info, this seems consistent with the idea that Connext is not parsing it properly and interpreting it as local time and not UTC.

If your clock was off, the date in the certificate will be off and the one extracted from the certificate and copied into the permission file would be off too. So I think it rules out a "time sync" issue.

---

We can find a nicer fix that this. I can try a thing or 2 tomorrow. If this fix the MacOS jobs I'm fine merging this and https://github.com/ros2/sros2/pull/205 for now